profiles/killbill/src/test/java/org/killbill/billing/jaxrs/TestCache.java

```diff
@@ -0,0 +1,131 @@
+
+        // get all caches per account level
+        final Ehcache accountRecordIdCache = cacheManager.getEhcache(CacheType.ACCOUNT_RECORD_ID.getCacheName());
+        final Ehcache accountImmutableCache = cacheManager.getEhcache(CacheType.ACCOUNT_IMMUTABLE.getCacheName());
+        final Ehcache accountBcdCache = cacheManager.getEhcache(CacheType.ACCOUNT_BCD.getCacheName());
+
+        // verify that they are not null and have the accountId stored as a key (the account created before)
+        Assert.assertNotNull(accountRecordIdCache);
+        Assert.assertNotNull(accountRecordIdCache.get(input.getAccountId().toString()));
+        Assert.assertNotNull(accountImmutableCache);
+        Assert.assertNotNull(accountImmutableCache.get(input.getAccountId()));
+        Assert.assertNotNull(accountBcdCache);
+        Assert.assertNotNull(accountBcdCache.get(input.getAccountId()));
+
+        // invalidate caches per account level by accountId
+        killBillClient.invalidateCacheByAccount(input.getAccountId().toString(), requestOptions);
+
+        // verify that now the caches don't have the accountId key stored
+        Assert.assertNull(accountRecordIdCache.get(input.getAccountId().toString()));
+        Assert.assertNull(accountImmutableCache.get(input.getAccountId()));
+        Assert.assertNull(accountBcdCache.get(input.getAccountId()));
+    }
+
+    @Test(groups = "slow", description = "Can Invalidate (clear) all Tenant Caches for current Tenant")
+    public void testInvalidateCacheByTenant() throws Exception {
+
+        createAccountWithPMBundleAndSubscriptionAndWaitForFirstInvoice();
+
+        // get all caches per tenant level
+        final Ehcache tenantRecordIdCache = cacheManager.getEhcache(CacheType.TENANT_RECORD_ID.getCacheName());
+        final Ehcache tenantPaymentStateMachineConfigCache = cacheManager.getEhcache(CacheType.TENANT_PAYMENT_STATE_MACHINE_CONFIG.getCacheName());
+        final Ehcache tenantCache = cacheManager.getEhcache(CacheType.TENANT.getCacheName());
+        final Ehcache tenantKvCache = cacheManager.getEhcache(CacheType.TENANT_KV.getCacheName());
+        final Ehcache tenantConfigCache = cacheManager.getEhcache(CacheType.TENANT_CONFIG.getCacheName());
+        final Ehcache tenantOverdueConfigCache = cacheManager.getEhcache(CacheType.TENANT_OVERDUE_CONFIG.getCacheName());
+        final Ehcache tenantCatalogCache = cacheManager.getEhcache(CacheType.TENANT_CATALOG.getCacheName());
+
+        DefaultTenant currentTenant = (DefaultTenant) tenantCache.get(tenantCache.getKeys().get(0)).getObjectValue();
+
+        // verify that they are not null and have the expected tenant information
+        Assert.assertNotNull(tenantRecordIdCache);
+        Assert.assertNotNull(tenantRecordIdCache.get(currentTenant.getId().toString()));
+        Assert.assertNotNull(tenantPaymentStateMachineConfigCache);
+        String tenantPaymentStateMachineConfigCacheKey = "PLUGIN_PAYMENT_STATE_MACHINE_noop::1";
```

Comment:
Each test regenerates its own tenant - I don't think you can expect the tenant record id to be `1` (try running it on H2 for example to see if it fails: `mvn test -Ph2`).